File: java/bundles/org.eclipse.set.model.siteplan/src/org/eclipse/set/model/siteplan/TrackClose.java
/**
 * Copyright (c) 2021 DB Netz AG and others.
 * 
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 */
package org.eclipse.set.model.siteplan;


/**
 * <!-- begin-user-doc -->
 * A representation of the model object '<em><b>Track Close</b></em>'.
 * <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * </p>
 * <ul>
 *   <li>{@link org.eclipse.set.model.siteplan.TrackClose#getTrackCloseType <em>Track Close Type</em>}</li>
 *   <li>{@link org.eclipse.set.model.siteplan.TrackClose#getPosition <em>Position</em>}</li>
 * </ul>
 *
 * @see org.eclipse.set.model.siteplan.SiteplanPackage#getTrackClose()
 * @model
 * @generated
 */
public interface TrackClose extends SiteplanObject {
	/**
	 * Returns the value of the '<em><b>Track Close Type</b></em>' attribute.
	 * The literals are from the enumeration {@link org.eclipse.set.model.siteplan.TrackCloseType}.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Track Close Type</em>' attribute.
	 * @see org.eclipse.set.model.siteplan.TrackCloseType
	 * @see #setTrackCloseType(TrackCloseType)
	 * @see org.eclipse.set.model.siteplan.SiteplanPackage#getTrackClose_TrackCloseType()
	 * @model
	 * @generated
	 */
	TrackCloseType getTrackCloseType();

	/**
	 * Sets the value of the '{@link org.eclipse.set.model.siteplan.TrackClose#getTrackCloseType <em>Track Close Type</em>}' attribute.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Track Close Type</em>' attribute.
	 * @see org.eclipse.set.model.siteplan.TrackCloseType
	 * @see #getTrackCloseType()
	 * @generated
	 */
	void setTrackCloseType(TrackCloseType value);

	/**
	 * Returns the value of the '<em><b>Position</b></em>' containment reference.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Position</em>' containment reference.
	 * @see #setPosition(Position)
	 * @see org.eclipse.set.model.siteplan.SiteplanPackage#getTrackClose_Position()
	 * @model containment="true"
	 * @generated
	 */
	Position getPosition();

	/**
	 * Sets the value of the '{@link org.eclipse.set.model.siteplan.TrackClose#getPosition <em>Position</em>}' containment reference.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Position</em>' containment reference.
	 * @see #getPosition()
	 * @generated
	 */
	void setPosition(Position value);

} // TrackClose

File: web/siteplan/src/model/TrackClose.ts
/**
 * Copyright (c) 2022 DB Netz AG and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 */

import { checkInstance } from '@/util/ObjectExtension'
import { defaultPositionObj, Position } from './Position'
import SiteplanObject, { defaultObjectColorObj } from './SiteplanObject'

export enum TrackCloseType {
  FrictionBufferStop = 'Bremsprellbock',
  FixedBufferStop = 'Festprellbock',
  HeadRamp = 'Kopframpe',
  ThresholCross = 'Schwellenkreuz',
  InfrastructureBorder = 'Infrastrukturgrenze',
  TurnTable = 'Drehscheibe',
  SlidingStage = 'Schiebebuehne',
  FerryDock = 'Faehranleger',
  Other = 'sonstige'
}

export default interface TrackClose extends SiteplanObject{
  guid: string
  trackCloseType: TrackCloseType
  position: Position
}

export function defaultTrackCloseObj (): TrackClose {
  return {
    guid: '0',
    trackCloseType: TrackCloseType.FixedBufferStop,
    position: defaultPositionObj(),
    objectColors: [defaultObjectColorObj()]
  }
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function isInstanceofTrackClose (obj: any): boolean {
  return checkInstance(obj, defaultTrackCloseObj())
}
